### DISCUSSION TITLE
Bao support of the SiP Service Call interface

### DISCUSSION THREAD

**Discussion by jdbl24** [bao-project/bao-hypervisor#195] [General]
Hello,
I have a question regarding how Bao handles the SMC calls related to the **SiP service call interface** on IMX8QM SoCs.

In Xen, these SMC SiP calls (for dom0) are forwarded to the firmware because many drivers need to issue such SMCs for managing hardware ( [xen/arm: Add imx8q{m,x} platform glue](https://patchew.org/Xen/20240328163351.64808-1-john.ernberg@actia.se/20240328163351.64808-2-john.ernberg@actia.se/) ).

I am aware that Bao does not rely on a privileged VM such as Xen, but it looks like [Bao does not support the SiP service call interface](https://github.com/bao-project/bao-hypervisor/blob/5116da5de5d34a08a2cc22813d571c6a552b56a4/src/arch/armv8/aborts.c#L76).

Does this not create limitations for running VMs with Bao on the IMX8QM?
This may also apply to other platforms.

I wonder if the SMC SiP call interface is necessary for the operation of some platforms or if dispensing with it does not create limitations.
If the SMC SiP call interface were necessary for the operation on certain SoCs, how would it be implemented in a hypervisor such as Bao since there is no "Trusted VM"?

Thank you in advance for your reply and have a nice day.

Kind regards,

Jean

**Comment by DavidMCerdeira** [comment]
Greetings,

> I wonder if the SMC SiP call interface is necessary for the operation of some platforms or if dispensing with it does not create limitations.

Bao's current approach is limiting, you are correct.

> If the SMC SiP call interface were necessary for the operation on certain SoCs, how would it be implemented in a hypervisor such as Bao since there is no "Trusted VM"?

We are aware we need to tackle support for platform specific features like these. We haven't looked deeply into this subject yet, so I can't say with certainty what is the best approach. From experience, it seems that for some cases validating the SMC SiP call parameters should be sufficient. In other cases access to SMC SiP interface could be used as a way for guest to interfere with each other, and so they cannot be safely supported. At this moment I don't think there's a need for a trusted VM, but again, we haven't looked deeply into this yet.

**Comment by josecm** [comment]
@jdbl24 Can I ask what specific kind of SiP services you need in your setup?

And as @DavidMCerdeira pointed out, it should be straightforward to pass through these calls for a specific guest, which I believe would in essence become privilege. This is the reason for my previous question. If any of these SiP calls allows the guest to modify platform-wide configurations the guest would become part of the TCB.

**Reply by jdbl24** [reply]
Thank you @josecm @DavidMCerdeira for your prompt reply.

I do not need specific SiP services; I was just curious about how the SiP interface could be handled in Bao.

In the past, I tried to get Xen running on a Toradex Apalis Board (IMX8QM-based), but I was unable to get a Dom0 running properly since the hypervisor did not forward the SMC SiP calls. When running Xen on both IMX8QM and ZCU104 platforms, I observed that the kernel drivers in Dom0 constantly issue SiP service calls. 
Both Jailhouse and Xen simply forward any SMC SIP call issued by the **privileged VM** (i.e. the Root Cell or Dom0) while rejecting any SMC SiP calls from the other VMs. For this, it is very convenient to rely on a privileged VM (despite the drawbacks regarding the TCB) for managing platform hardware controls while ensuring that an untrusted VM cannot leverage sensitive functions exposed through the SMC SiP interface.

Thus, I was curious how this could be handled in a hypervisor like Bao, which does not rely on a privileged VM. If those SMCs are necessary for some platform specific controls, this inevitably impacts the TCB of the VM running on Bao as well. As @DavidMCerdeira  said, the best approach in my opinion would be to implement some access control on the SMC interface but I am not sure it is always  sufficient.

**Reply by josecm** [reply]
> As @DavidMCerdeira said, the best approach in my opinion would be to implement some access control on the SMC interface but I am not sure it is always sufficient.

It is our goal to implement these kind of platform-specific checks both for system service hypercalls and other plaform-wide configuration peripherals (e.g., clocks, power, etc). However, it is not in our immediate pipeline...

Why you don't think this might be sufficient?

**Reply by PendemRani** [reply]
Hi @jdbl24, 
I am facing an SMC call issue in imx8QM and as well as in imx8QXPMEK while dom0 kernel is booting after the xen loading, do you have any idea how to resolve this issue?
**(XEN) d0v0 Unhandled SMC/HVC: 0xc6000000
[    1.458276] Error FSL_HVC_SC -1**

These are the logs:
(XEN) CPU0: Guest atomics will try 11 times before pausing the domain
(XEN) Bringing up CPU1
(XEN) GICv3: CPU1: Found redistributor in region 0 @00000a004003c000
(XEN) CPU1: Guest atomics will try 12 times before pausing the domain
(XEN) CPU 1 booted.
(XEN) Bringing up CPU2
(XEN) GICv3: CPU2: Found redistributor in region 0 @00000a004005c000
(XEN) CPU2: Guest atomics will try 11 times before pausing the domain
(XEN) CPU 2 booted.
(XEN) Bringing up CPU3
(XEN) GICv3: CPU3: Found redistributor in region 0 @00000a004007c000
(XEN) CPU3: Guest atomics will try 11 times before pausing the domain
(XEN) Brought up 4 CPUs
(XEN) CPU 3 booted.
(XEN) I/O virtualisation disabled
(XEN) P2M: 40-bit IPA with 40-bit PA and 8-bit VMID
(XEN) P2M: 3 levels with order-1 root, VTCR 0x0000000080023558
(XEN) Scheduling granularity: cpu, 1 CPU per sched-resource
(XEN) Initializing null scheduler
(XEN) WARNING: This is experimental software in development.
(XEN) Use at your own risk.
(XEN) alternatives: Patching with alt table 00000a00002ee090 -> 00000a00002ef2cc
(XEN) *** LOADING DOMAIN 0 ***
(XEN) Loading d0 kernel from boot module @ 000000009e000000
(XEN) Allocating 1:1 mappings totalling 1512MB for dom0:
(XEN) BANK[0] 0x000000a8000000-0x000000f8000000 (1280MB)
(XEN) BANK[1] 0x000008aa000000-0x000008b8000000 (224MB)
(XEN) BANK[2] 0x000008bf000000-0x000008bf800000 (8MB)
(XEN) Grant table range: 0x00000080200000-0x00000080240000
(XEN) Allocating PPI 16 for event channel interrupt
(XEN) d0: extended region 0: 0x94c00000->0xa7e00000
(XEN) d0: extended region 1: 0xf8000000->0xfde00000
(XEN) d0: extended region 2: 0x880000000->0x8a9e00000
(XEN) d0: extended region 3: 0x8b8000000->0x8bee00000
(XEN) Loading zImage from 000000009e000000 to 00000000a8000000-00000000aa15ba00
(XEN) Loading d0 DTB to 0x00000000b0000000-0x00000000b001e709
(XEN) Initial low memory virq threshold set at 0x4000 pages.
(XEN) Std. Loglevel: All
(XEN) Guest Loglevel: All
(XEN) *** Serial input to DOM0 (type 'CTRL-a' three times to switch input)
(XEN) xen-source/xen/common/sched/null.c:357: 0 <-- d0v0
(XEN) Freed 408kB init memory.
(XEN) d0v0 Unhandled SMC/HVC: 0x84000050
(XEN) d0v0 Unhandled SMC/HVC: 0x8600ff01
(XEN) d0v0: vGICD: RAZ on reserved register offset 0x00000c
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER4
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER8
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER12
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER16
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER20
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER24
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER28
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER32
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER36
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER40
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER44
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER48
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER52
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER56
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER60
(XEN) d0v0: vGICD: unhandled word write 0x000000ffffffff to ICACTIVER64
(XEN) d0v0: vGICR: SGI: unhandled word write 0x000000ffffffff to ICACTIVER0
[    0.000000] Booting Linux on physical CPU 0x0000000000 [0x410fd042]
[    0.000000] Linux version 6.12.3-lts-next-g37d02f4dcbbe (oe-user@oe-host) (aarch64-poky-linux-gcc (GCC) 14.2.0, GNU ld (GNU Binutils) 2.43.1) #1 SMP PREEMPT Sat Mar  1 12:54:50 UTC 2025
[    0.000000] KASLR disabled due to lack of seed
[    0.000000] Machine model: Freescale i.MX8QXP MEK
[    0.000000] Xen 4.20 support found
[    0.000000] efi: UEFI not found.
[    0.000000] OF: reserved mem: failed to allocate memory for node 'linux,cma': size 960 MiB
[    0.000000] OF: reserved mem: 0x0000000084000000..0x0000000085ffffff (32768 KiB) nomap non-reusable decoder-boot@84000000
[    0.000000] OF: reserved mem: 0x0000000086000000..0x00000000861fffff (2048 KiB) nomap non-reusable encoder-boot@86000000
[    0.000000] OF: reserved mem: 0x0000000090000000..0x0000000090007fff (32 KiB) nomap non-reusable vdev0vring0@90000000
[    0.000000] OF: reserved mem: 0x0000000090008000..0x000000009000ffff (32 KiB) nomap non-reusable vdev0vring1@90008000
[    0.000000] OF: reserved mem: 0x0000000090010000..0x0000000090017fff (32 KiB) nomap non-reusable vdev1vring0@90010000
[    0.000000] OF: reserved mem: 0x0000000090018000..0x000000009001ffff (32 KiB) nomap non-reusable vdev1vring1@90018000
[    0.000000] OF: reserved mem: 0x00000000900ff000..0x00000000900fffff (4 KiB) nomap non-reusable rsc-table@900ff000
[    0.000000] Reserved memory: created DMA memory pool at 0x0000000090400000, size 1 MiB
[    0.000000] OF: reserved mem: initialized node vdevbuffer@90400000, compatible id shared-dma-pool
[    0.000000] OF: reserved mem: 0x0000000090400000..0x00000000904fffff (1024 KiB) nomap non-reusable vdevbuffer@90400000
[    0.000000] OF: reserved mem: 0x0000000092000000..0x00000000920fffff (1024 KiB) nomap non-reusable decoder-rpc@92000000
[    0.000000] OF: reserved mem: 0x0000000092400000..0x00000000933fffff (16384 KiB) nomap non-reusable dsp@92400000
[    0.000000] OF: reserved mem: 0x0000000093400000..0x00000000942effff (15296 KiB) nomap non-reusable dsp_reserved_heap@93400000
[    0.000000] OF: reserved mem: 0x00000000942f0000..0x00000000942f7fff (32 KiB) nomap non-reusable vdev0vring0@942f0000
[    0.000000] OF: reserved mem: 0x00000000942f8000..0x00000000942fffff (32 KiB) nomap non-reusable vdev0vring1@942f8000
[    0.000000] Reserved memory: created DMA memory pool at 0x0000000094300000, size 1 MiB
[    0.000000] OF: reserved mem: initialized node vdev0buffer@94300000, compatible id shared-dma-pool
[    0.000000] OF: reserved mem: 0x0000000094300000..0x00000000943fffff (1024 KiB) nomap non-reusable vdev0buffer@94300000
[    0.000000] OF: reserved mem: 0x0000000094400000..0x0000000094afffff (7168 KiB) nomap non-reusable encoder-rpc@94400000
[    0.000000] OF: reserved mem: 0x00000000fe000000..0x00000000ffdfffff (30720 KiB) nomap non-reusable optee_core@fe000000
[    0.000000] OF: reserved mem: 0x00000000ffe00000..0x00000000ffffffff (2048 KiB) nomap non-reusable optee_shm@ffe00000
[    0.000000] NUMA: Faking a node at [mem 0x0000000084000000-0x00000008bf7fffff]
[    0.000000] NODE_DATA(0) allocated [mem 0x8bf47fa40-0x8bf48247f]
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000084000000-0x00000000ffffffff]
[    0.000000]   DMA32    empty
[    0.000000]   Normal   [mem 0x0000000100000000-0x00000008bf7fffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000084000000-0x00000000861fffff]
[    0.000000]   node   0: [mem 0x0000000090000000-0x000000009001ffff]
[    0.000000]   node   0: [mem 0x00000000900ff000-0x00000000900fffff]
[    0.000000]   node   0: [mem 0x0000000090400000-0x00000000904fffff]
[    0.000000]   node   0: [mem 0x0000000092000000-0x00000000920fffff]
[    0.000000]   node   0: [mem 0x0000000092400000-0x0000000094afffff]
[    0.000000]   node   0: [mem 0x00000000a8000000-0x00000000f7ffffff]
[    0.000000]   node   0: [mem 0x00000000fe000000-0x00000000ffffffff]
[    0.000000]   node   0: [mem 0x00000008aa000000-0x00000008b7ffffff]
[    0.000000]   node   0: [mem 0x00000008bf000000-0x00000008bf7fffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000084000000-0x00000008bf7fffff]
[    0.000000] On node 0, zone DMA: 16384 pages in unavailable ranges
[    0.000000] On node 0, zone DMA: 7680 pages in unavailable ranges
[    0.000000] On node 0, zone DMA: 223 pages in unavailable ranges
[    0.000000] On node 0, zone DMA: 768 pages in unavailable ranges
[    0.000000] On node 0, zone DMA: 6912 pages in unavailable ranges
[    0.000000] On node 0, zone DMA: 768 pages in unavailable ranges
[    0.000000] On node 0, zone DMA: 13568 pages in unavailable ranges
[    0.000000] On node 0, zone DMA: 24576 pages in unavailable ranges
[    0.000000] On node 0, zone Normal: 8192 pages in unavailable ranges
[    0.000000] On node 0, zone Normal: 28672 pages in unavailable ranges
[    0.000000] On node 0, zone Normal: 2048 pages in unavailable ranges
[    0.000000] cma: Reserved 32 MiB at 0x00000000f6000000 on node -1
[    0.000000] psci: probing for conduit method from DT.
[    0.000000] psci: PSCIv1.1 detected in firmware.
[    0.000000] psci: Using standard PSCI v0.2 function IDs
[    0.000000] psci: Trusted OS migration not required
[    0.000000] psci: SMC Calling Convention v1.2
[    0.000000] percpu: Embedded 25 pages/cpu s61528 r8192 d32680 u102400
[    0.000000] Detected VIPT I-cache on CPU0
[    0.000000] CPU features: detected: GIC system register CPU interface
[    0.000000] alternatives: applying boot alternatives
[    0.000000] Kernel command line: console=hvc0 earlycon=xen earlyprintk=xen clk_ignore_unused root=/dev/mmcblk1p2 rw rootwait
[    0.000000] Unknown kernel command line parameters "earlyprintk=xen", will be passed to user space.
[    0.000000] Dentry cache hash table entries: 262144 (order: 9, 2097152 bytes, linear)
[    0.000000] Inode-cache hash table entries: 131072 (order: 8, 1048576 bytes, linear)
[    0.000000] Fallback order for Node 0: 0 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 414497
[    0.000000] Policy zone: Normal
[    0.000000] mem auto-init: stack:all(zero), heap alloc:off, heap free:off
[    0.000000] software IO TLB: area num 1.
[    0.000000] software IO TLB: mapped [mem 0x00000000f2000000-0x00000000f6000000] (64MB)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] rcu: Preemptible hierarchical RCU implementation.
[    0.000000] rcu: 	RCU event tracing is enabled.
[    0.000000] rcu: 	RCU restricting CPUs from NR_CPUS=512 to nr_cpu_ids=1.
[    0.000000] 	Trampoline variant of Tasks RCU enabled.
[    0.000000] 	Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 25 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=1
[    0.000000] RCU Tasks: Setting shift to 0 and lim to 1 rcu_task_cb_adjust=1 rcu_task_cpu_ids=1.
[    0.000000] RCU Tasks Trace: Setting shift to 0 and lim to 1 rcu_task_cb_adjust=1 rcu_task_cpu_ids=1.
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] GICv3: 512 SPIs implemented
[    0.000000] GICv3: 0 Extended SPIs implemented
[    0.000000] Root IRQ handler: gic_handle_irq
[    0.000000] GICv3: GICv3 features: 16 PPIs
[    0.000000] GICv3: GICD_CTRL.DS=0, SCR_EL3.FIQ=0
[    0.000000] GICv3: CPU0: found redistributor 0 region 0:0x0000000051b00000
[    0.000000] rcu: srcu_init: Setting srcu_struct sizes based on contention.
[    0.000000] arch_timer: cp15 timer(s) running at 8.00MHz (virt).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x1d854df40, max_idle_ns: 440795202120 ns
[    0.000001] sched_clock: 56 bits at 8MHz, resolution 125ns, wraps every 2199023255500ns
[    0.001049] Console: colour dummy device 80x25
[    0.001070] printk: legacy console [hvc0] enabled
[    0.778603] Calibrating delay loop (skipped), value calculated using timer frequency.. 16.00 BogoMIPS (lpj=32000)
[    0.789928] pid_max: default: 32768 minimum: 301
[    0.795164] LSM: initializing lsm=capability
[    0.800061] Mount-cache hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.808228] Mountpoint-cache hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.818104] cacheinfo: Unable to detect cache hierarchy for CPU 0
[    0.826871] xen:grant_table: Grant tables using version 1 layout
[    0.833582] Grant table initialized
[    0.837446] xen:events: Using FIFO-based ABI
[    0.842247] Xen: initializing cpu0
[    0.846307] rcu: Hierarchical SRCU implementation.
[    0.851590] rcu: 	Max phase no-delay instances is 1000.
[    0.858911] EFI services will not be available.
[    0.864167] smp: Bringing up secondary CPUs ...
[    0.869202] smp: Brought up 1 node, 1 CPU
[    0.873680] SMP: Total of 1 processors activated.
[    0.878934] CPU: All CPU(s) started at EL1
[    0.883553] CPU features: detected: 32-bit EL0 Support
[    0.889233] CPU features: detected: 32-bit EL1 Support
[    0.894970] CPU features: detected: CRC32 instructions
[    0.900754] alternatives: applying system-wide alternatives
[    0.908391] Memory: 1372884K/1657988K available (22016K kernel code, 1646K rwdata, 8244K rodata, 2112K init, 778K bss, 250684K reserved, 32768K cma-reserved)
[    0.924470] devtmpfs: initialized
[    0.946798] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    0.957589] futex hash table entries: 256 (order: 2, 16384 bytes, linear)
[    0.966215] 24016 pages in range for non-PLT usage
[    0.966241] 515536 pages in range for PLT usage
[    0.971878] pinctrl core: initialized pinctrl subsystem
[    0.988000] DMI not present or invalid.
[    0.995048] NET: Registered PF_NETLINK/PF_ROUTE protocol family
[    1.002599] DMA: preallocated 256 KiB GFP_KERNEL pool for atomic allocations
[    1.010476] DMA: preallocated 256 KiB GFP_KERNEL|GFP_DMA pool for atomic allocations
[    1.019133] DMA: preallocated 256 KiB GFP_KERNEL|GFP_DMA32 pool for atomic allocations
[    1.027914] audit: initializing netlink subsys (disabled)
[    1.034107] audit: type=2000 audit(0.540:1): state=initialized audit_enabled=0 res=1
[    1.043719] thermal_sys: Registered thermal governor 'step_wise'
[    1.043730] thermal_sys: Registered thermal governor 'power_allocator'
[    1.050721] hw-breakpoint: found 6 breakpoint and 4 watchpoint registers.
[    1.065537] ASID allocator initialised with 65536 entries
[    1.071947] Serial: AMBA PL011 UART driver
[    1.076569] imx mu driver is registered.
[    1.080930] imx rpmsg driver is registered.
[    1.140357] platform 56180000.dpu: Fixed dependency cycle(s) with /bus@56220000/dsi_host@56228000
[    1.150266] platform 56228000.dsi_host: Fixed dependency cycle(s) with /bus@56220000/i2c@56226000/adv7535@3d
[    1.161139] platform 56228000.dsi_host: Fixed dependency cycle(s) with /bus@56000000/dpu@56180000
[    1.172646] platform 56180000.dpu: Fixed dependency cycle(s) with /bus@56220000/dsi_host@56248000
[    1.182522] platform 56248000.dsi_host: Fixed dependency cycle(s) with /bus@56220000/i2c@56246000/adv7535@3d
[    1.193402] platform 56248000.dsi_host: Fixed dependency cycle(s) with /bus@56000000/dpu@56180000
[    1.209923] HugeTLB: registered 1.00 GiB page size, pre-allocated 0 pages
[    1.217501] HugeTLB: 0 KiB vmemmap can be freed for a 1.00 GiB page
[    1.224441] HugeTLB: registered 32.0 MiB page size, pre-allocated 0 pages
[    1.231974] HugeTLB: 0 KiB vmemmap can be freed for a 32.0 MiB page
[    1.238941] HugeTLB: registered 2.00 MiB page size, pre-allocated 0 pages
[    1.246493] HugeTLB: 0 KiB vmemmap can be freed for a 2.00 MiB page
[    1.253465] HugeTLB: registered 64.0 KiB page size, pre-allocated 0 pages
[    1.261012] HugeTLB: 0 KiB vmemmap can be freed for a 64.0 KiB page
[    1.272246] ACPI: Interpreter disabled.
[    1.277958] xen:balloon: Initialising balloon driver
[    1.286451] iommu: Default domain type: Translated
[    1.291833] iommu: DMA domain TLB invalidation policy: strict mode
[    1.299065] SCSI subsystem initialized
[    1.303703] usbcore: registered new interface driver usbfs
[    1.309785] usbcore: registered new interface driver hub
[    1.315712] usbcore: registered new device driver usb
[    1.324202] mc: Linux media interface: v0.10
[    1.329013] videodev: Linux video capture interface: v2.00
[    1.335153] pps_core: LinuxPPS API ver. 1 registered
[    1.340627] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    1.350781] PTP clock support registered
[    1.355197] EDAC MC: Ver: 3.0.0
[    1.359177] scmi_core: SCMI protocol bus registered
[    1.365444] FPGA manager framework
[    1.369335] Advanced Linux Sound Architecture Driver Initialized.
[    1.376952] Bluetooth: Core ver 2.22
[    1.380937] NET: Registered PF_BLUETOOTH protocol family
[    1.386828] Bluetooth: HCI device and connection manager initialized
[    1.393905] Bluetooth: HCI socket layer initialized
[    1.399349] Bluetooth: L2CAP socket layer initialized
[    1.405014] Bluetooth: SCO socket layer initialized
[    1.411333] vgaarb: loaded
(XEN) d0v0 Unhandled SMC/HVC: 0xc6000000
[    1.419022] Error FSL_HVC_SC -1
[    1.422531] imx_scu_soc_id: get soc info failed, ret -5
[    1.428375] imx-scu system-controller: failed to initialize SoC info: -22
[    1.435950] imx-scu system-controller: NXP i.MX SCU Initialized
(XEN) d0v0 Unhandled SMC/HVC: 0xc6000000
[    1.458276] Error FSL_HVC_SC -1
(XEN) d0v0 Unhandled SMC/HVC: 0xc6000000
[    1.465791] Error FSL_HVC_SC -1
[    1.469319] imx-scu-pd system-controller:power-controller: failed to get power resource 191 mode, ret -5
(XEN) d0v0 Unhandled SMC/HVC: 0xc6000000
[    1.484517] Error FSL_HVC_SC -1
(XEN) d0v0 Unhandled SMC/HVC: 0xc6000000
[    1.492065] Error FSL_HVC_SC -1
[    1.495591] imx-scu-pd system-controller:power-controller: failed to get power resource 192 mode, ret -5
(XEN) d0v0 Unhandled SMC/HVC: 0xc6000000
[    1.510128] Error FSL_HVC_SC -1
(XEN) d0v0 Unhandled SMC/HVC: 0xc6000000
[    1.517676] Error FSL_HVC_SC -1
[    1.521202] imx-scu-pd system-controller:power-controller: failed to get power resource 193 mode, ret -5

**Reply by josecm** [reply]
@PendemRani  I am confused as why you are sharing a XEN log on another hypervisor discussion page.
